bionic/protocols.py

```diff
@@ -86,15 +122,22 @@ def read(self, path):
     }
 
     def tokenize(self, value):
+        """
+        Converts a valid value into a unique hash which can be used as a filename.
+        """
         if type(value) in self.SIMPLE_TYPES:
             return tokenization.tokenize(value)
         else:
             return tokenization.tokenize(value, self._write_to_bytes)
 
     def tokenize_file(self, path):
+        """Like ``tokenize``, but operates on a serialized value."""
```

Comment:
Can we be more specific here that the serialized value is coming from the contents of the path file (or dir)?

Reply:
Related to this, even I have trouble remembering if this path is a pathlib path. I remember seeing [typeguard](https://github.com/agronholm/typeguard) before for this, how do you feel about using for at least paths in bionic?